tests/unit/compiler/venom/test_mem_ssa.py

```diff
@@ -0,0 +1,1249 @@
+            stop
+    }
+    """
+
+    mem_ssa, fn, _ = create_mem_ssa(pre)
+
+    block1 = fn.get_basic_block("block1")
+    block2 = fn.get_basic_block("block2")
+    merge_block = fn.get_basic_block("merge")
+
+    def1 = mem_ssa.get_memory_def(block1.instructions[0])  # mstore 0, 42
+    def2 = mem_ssa.get_memory_def(block2.instructions[0])  # mstore 0, 24
+    def3 = mem_ssa.get_memory_def(merge_block.instructions[0])  # mstore 0, 84
+
+    assert merge_block in mem_ssa.memory_phis, "Merge block should have a phi node"
+    phi = mem_ssa.memory_phis[merge_block]
+
+    assert len(phi.operands) == 2, "Phi node should have 2 operands"
+    assert phi.operands[0][0] == def1, "First operand should be def1"
+    assert phi.operands[1][0] == def2, "Second operand should be def2"
+    assert phi.operands[0][1] == block1, "First operand should be from block1"
+    assert phi.operands[1][1] == block2, "Second operand should be from block2"
+
+    assert (
+        def3.reaching_def == mem_ssa.live_on_entry
+    ), "def3's reaching definition should be live_on_entry"
+
+    # Create a new memory definition with the same location as def3
+    new_def = MemoryDef(mem_ssa.next_id, merge_block.instructions[0])
+    mem_ssa.next_id += 1
+    new_def.loc = def3.loc
+
+    # Manually test the _get_reaching_def_for_def method
+    reaching_def = mem_ssa._get_reaching_def_for_def(new_def)
+    assert reaching_def == phi, "The reaching definition should be the phi node"
+
+
+def test_memory_access_properties():
+    live_access = MemoryAccess(0)
+    assert live_access.is_live_on_entry
+    assert not live_access.is_volatile
+    assert live_access.id_str == "live_on_entry"
+
+    regular_access = MemoryAccess(1)
+    assert not regular_access.is_live_on_entry
+    assert regular_access.id_str == "1"
+
+    another_access = MemoryAccess(1)
+    assert regular_access == another_access
+    assert hash(regular_access) == hash(another_access)
+    assert regular_access != live_access
+    assert regular_access != "not_a_memory_access"
+
+
+def test_mark_location_volatile(create_mem_ssa):
+    pre = """
+    function _global {
+        entry:
+            mstore 0, 42
+            mstore 32, 24
+            stop
+    }
+    """
+    mem_ssa, fn, _ = create_mem_ssa(pre)
+
+    bb = fn.get_basic_block("entry")
+    def1 = mem_ssa.get_memory_def(bb.instructions[0])  # mstore 0, 42
+    def2 = mem_ssa.get_memory_def(bb.instructions[1])  # mstore 32, 24
+
+    # Mark first location as volatile
+    volatile_loc = mem_ssa.mark_location_volatile(def1.loc)
+    assert volatile_loc.is_volatile
+    assert def1.loc.is_volatile
+    assert not def2.loc.is_volatile
+
+
+def test_remove_redundant_phis(create_mem_ssa):
+    pre = """
+    function _global {
+        entry:
+            %cond = 1
+            jnz %cond, @block1, @block2
+        block1:
+            mstore 0, 42
+            jmp @merge
+        block2:
+            mstore 0, 42  # Same value as block1
```

Comment:
doesn't need to be the same value, for instance `mstore 0, 41` also passes the test